docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -1,420 +1,208 @@
+:cairo-selectors: https://github.com/starkware-libs/cairo/blob/7dd34f6c57b7baf5cd5a30c15e00af39cb26f7e1/crates/cairo-lang-starknet/src/contract.rs#L39-L48[Cairo]
+:solidity-selectors: https://solidity-by-example.org/function-selector/[Solidity]
+
+Although Starknet is not EVM compatible, this implementation aims to be as close as possible to the ERC20 standard, in the following ways:
 
-* It uses Cairo's `uint256` instead of `felt`.
-* It returns `TRUE` as success.
-* It accepts a `felt` argument for `decimals` in the constructor calldata with a max value of 2{caret}8 (imitating `uint8` type).
 * It makes use of Cairo's short strings to simulate `name` and `symbol`.
 
 But some differences can still be found, such as:
 
-* `transfer`, `transferFrom` and `approve` will never return anything different from `TRUE` because they will revert on any error.
-* Function selectors are calculated differently between https://github.com/starkware-libs/cairo-lang/blob/7712b21fc3b1cb02321a58d0c0579f5370147a8b/src/starkware/starknet/public/abi.py#L25[Cairo] and https://solidity-by-example.org/function-selector/[Solidity].
+* Function selectors are written in snake_case as opposed to camelCase in Solidity.
+* `transfer`, `transfer_from` and `approve` will never return anything different from `true` because they will revert on any error.
+* Function selectors are calculated differently between {cairo-selectors} and {solidity-selectors}.
 
 == Usage
 
-Use cases go from medium of exchange currency to voting rights, staking, and more.
-
-Considering that the constructor method looks like this:
+:components: https://community.starknet.io/t/cairo-1-contract-syntax-is-evolving/94794#extensibility-and-components-11[Components]
+:erc20-supply: xref:/guides/erc20-supply.adoc[Creating ERC20 Supply]
 
-[,cairo]
-----
-func constructor(
-    name: felt,               // Token name as Cairo short string
-    symbol: felt,             // Token symbol as Cairo short string
-    decimals: felt            // Token decimals (usually 18)
-    initial_supply: Uint256,  // Amount to be minted
-    recipient: felt           // Address where to send initial supply to
-) {
-}
-----
-
-To create a token you need to deploy it like this:
-
-[,python]
-----
-erc20 = await starknet.deploy(
-    "openzeppelin/token/erc20/presets/ERC20.cairo",
-    constructor_calldata=[
-        str_to_felt("Token"),     # name
-        str_to_felt("TKN"),       # symbol
-        18,                       # decimals
-        (1000, 0),                # initial supply
-        account.contract_address  # recipient
-    ]
-)
-----
+WARNING: The following example uses a contract's `unsafe_new_contract_state` to access another contract's state.
+This is currently unsafe, because storage members could clash among used contracts if not reviewed carefully.
+Extensibility will be revisited after {components} are introduced.
```

Comment:
```suggestion
WARNING: The following example uses `unsafe_new_contract_state` to access another contract's state.
Although this is useful to use them as modules, it's considered unsafe because storage members could clash among used contracts if not reviewed carefully.
Extensibility will be revisited after {components} are introduced.
```
i think this reads better. that being said, is it really unsafe? if we pass `src5_storage()` to `SRC5` functions, is it problematic? I think the issue arises if there's storage collisions among different used modules, but that's mitigated by prefixes. either way, it's not related to `unsafe_new_contract_state`, isn't the same with components?

Reply:
Great questions! The warning was added a while ago. As you mentioned, prefixes mitigate the potential for storage clashes. AFAIK that's what's unsafe about it. I'll remove the warning

Until we have components, though, I don't think it's a bad idea to inform users of this...it probably belongs in the forthcoming `Extensibility` doc

Reply:
sure we can inform about this. what do you think about the suggestion?

Reply:
I think the suggestion indeed reads better!

Just to clarify, do we agree to remove the warning now and insert it into `Extensibility` in the future? If not, should we include the warning in each doc `Usage` section for consistency?

Reply:
ideally we would add a single explanation in Extensibility and then link there from every section, but i wonder whether it makes sense to write such a page for unsafe storage usage when components are around the corner

Reply:
Hmm right. I don't like the idea of assuming components will arrive soon. How's this: let's make a tentative plan to include an unsafe storage section in Extensibility. If we get components sooner, we'll have to make adjustments to Extensibility and the other docs anyway. If we don't have components by the time the other docs are done, we can just carry on until tenemos componentes